diff --git a/modules/mod-playerbots/src/playerbot/WorldPosition.cpp b/modules/mod-playerbots/src/playerbot/WorldPosition.cpp
--- a/modules/mod-playerbots/src/playerbot/WorldPosition.cpp
+++ b/modules/mod-playerbots/src/playerbot/WorldPosition.cpp
@@ -12,9 +12,9 @@
 
     if (transports.empty() || !entry) //Elevators&rams
     {
         for (auto gopair : getGameObjectsNear(0.0f, entry))
-            if (GameObject* go = map->GetGameObject(gopair->first))
+            if (GameObject* go = map->GetGameObject(ObjectGuid(HIGHGUID_GAMEOBJECT, gopair->second.id, gopair->first)))
                 if (GenericTransport* transport = dynamic_cast<GenericTransport*>(go))
                     transports.insert(transport);
     }
 
