diff --git a/dep/include/g3dlite/G3D/platform.h b/dep/include/g3dlite/G3D/platform.h
--- a/dep/include/g3dlite/G3D/platform.h
+++ b/dep/include/g3dlite/G3D/platform.h
@@ -2,5 +2,6 @@
     #define G3D_WINDOWS
     #undef __MSVCRT_VERSION__
     #define __MSVCRT_VERSION__ 0x0601
+    #include <winsock2.h>
     #include <windows.h>
 #elif  defined(__FreeBSD__) || defined(__OpenBSD__)
diff --git a/dep/src/g3dlite/debugAssert.cpp b/dep/src/g3dlite/debugAssert.cpp
--- a/dep/src/g3dlite/debugAssert.cpp
+++ b/dep/src/g3dlite/debugAssert.cpp
@@ -1,4 +1,4 @@
         // Make sure the message got translated into something.
-        LPTSTR realLastErr;
+        LPCTSTR realLastErr;
         if (NULL != formatMsg) {
             realLastErr = formatMsg;
diff --git a/dep/src/g3dlite/FileSystem.cpp b/dep/src/g3dlite/FileSystem.cpp
--- a/dep/src/g3dlite/FileSystem.cpp
+++ b/dep/src/g3dlite/FileSystem.cpp
@@ -1,4 +1,7 @@
 #  ifdef __MINGW32__
+#    ifdef stat64
+#      undef stat64
+#    endif
 #    define stat64 stat
 #  else
 #    define stat64 _stat64
