diff --git a/src/game/Spells/SpellEffects.cpp b/src/game/Spells/SpellEffects.cpp
index f4cb139..cdd8f4b 100644
--- a/src/game/Spells/SpellEffects.cpp
+++ b/src/game/Spells/SpellEffects.cpp
@@ -2269,7 +2269,7 @@ void Spell::EffectDispel(SpellEffectIndex eff_idx)
 
     // Fill possible dispel list
     int32 priority_dispel = -1;
-    std::list <std::pair<SpellAuraHolder* , uint32> > dispel_list;
+    std::vector<std::pair<SpellAuraHolder*, uint32>> dispel_list;
 
     bool checkFaction = true;
     // Pierre de sort dissipe sorts negatifs et positifs.
@@ -2280,6 +2280,7 @@ void Spell::EffectDispel(SpellEffectIndex eff_idx)
     int32 dispel_type = m_spellInfo->EffectMiscValue[eff_idx];
     uint32 dispelMask  = GetDispellMask(dispel_type < 0 ? DISPEL_ALL : DispelType(dispel_type));
     Unit::SpellAuraHolderMap const& auras = unitTarget->GetSpellAuraHolderMap();
+    dispel_list.reserve(auras.size());
     for (const auto& aura : auras)
     {
         SpellAuraHolder* holder = aura.second;
@@ -2322,15 +2323,15 @@ void Spell::EffectDispel(SpellEffectIndex eff_idx)
         for (int32 count = 0; count < damage && !dispel_list.empty(); ++count)
         {
             // Random select buff for dispel
-            std::list<std::pair<SpellAuraHolder*, uint32> >::iterator dispel_itr = dispel_list.begin();
+            auto dispel_itr = dispel_list.begin();
             if (priority_dispel >= 0)
             {
-                std::advance(dispel_itr, priority_dispel);
+                dispel_itr += priority_dispel;
                 priority_dispel = -1;
             }
             else
             {
-                std::advance(dispel_itr, urand(0, dispel_list.size() - 1));
+                dispel_itr += urand(0, dispel_list.size() - 1);
             }
 
             SpellAuraHolder *holder = dispel_itr->first;
