diff --git a/src/game/Handlers/LootHandler.cpp b/src/game/Handlers/LootHandler.cpp
index c9cf32a..246484d 100644
--- a/src/game/Handlers/LootHandler.cpp
+++ b/src/game/Handlers/LootHandler.cpp
@@ -361,6 +361,7 @@ void WorldSession::HandleLootMoneyOpcode(WorldPacket & /*recv_data*/)
             Group *group = player->GetGroup();
 
             std::vector<Player*> playersNear;
+            playersNear.reserve(group->GetMembersCount());
             for (GroupReference *itr = group->GetFirstMember(); itr != nullptr; itr = itr->next())
             {
                 Player* playerGroup = itr->getSource();
