diff --git a/src/game/Spells/SpellEffects.cpp b/src/game/Spells/SpellEffects.cpp
index cdd8f4b..9d12e69 100644
--- a/src/game/Spells/SpellEffects.cpp
+++ b/src/game/Spells/SpellEffects.cpp
@@ -3939,6 +3939,7 @@ void Spell::EffectScriptEffect(SpellEffectIndex eff_idx)
                     // Select maintank + 4 random targets
                     std::vector<Unit*> viableTargets;
                     const ThreatList& tl = m_casterUnit->GetThreatManager().getThreatList();
+                    viableTargets.reserve(tl.size());
                     for (const auto it : tl)
                     {
                         if (it->getUnitGuid().IsPlayer())
