diff --git a/dep/dpp/dpp/export.h b/dep/dpp/dpp/export.h
index 423729c..28dc1f1 100644
--- a/dep/dpp/dpp/export.h
+++ b/dep/dpp/dpp/export.h
@@ -62,5 +62,5 @@
 #ifndef _WIN32
 	#define SOCKET int
 #else
-	#include <WinSock2.h>
+    #include <winsock2.h>
 #endif
\ No newline at end of file
diff --git a/src/game/Maps/GridDefines.h b/src/game/Maps/GridDefines.h
index 1798cbc..60ad825 100644
--- a/src/game/Maps/GridDefines.h
+++ b/src/game/Maps/GridDefines.h
@@ -185,7 +185,7 @@ namespace MaNGOS
     }
     inline bool IsValidMapCoord(float c)
     {
-        return finite(c) && (std::fabs(c) <= MAP_HALFSIZE - 0.5);
+        return std::isfinite(c) && (std::fabs(c) <= MAP_HALFSIZE - 0.5);
     }
 
     inline bool IsValidMapCoord(float x, float y)
@@ -200,7 +200,7 @@ namespace MaNGOS
 
     inline bool IsValidMapCoord(float x, float y, float z, float o)
     {
-        return IsValidMapCoord(x,y,z) && finite(o) && fabs(o) <= 4 * M_PI;
+        return IsValidMapCoord(x,y,z) && std::isfinite(o) && fabs(o) <= 4 * M_PI;
     }
 }
 #endif
diff --git a/src/shared/ByteBuffer.h b/src/shared/ByteBuffer.h
index fd75111..d1974e9 100644
--- a/src/shared/ByteBuffer.h
+++ b/src/shared/ByteBuffer.h
@@ -121,7 +121,7 @@ class ByteBuffer
             return *this;
         }
 
-#ifdef __MINGW32__
+#if defined(__MINGW32__) && !defined(__MINGW64__)
         ByteBuffer &operator<<(time_t value)
         {
             append<time_t>(value);
@@ -217,7 +217,7 @@ class ByteBuffer
             return *this;
         }
 
-#ifdef __MINGW32__
+#if defined(__MINGW32__) && !defined(__MINGW64__)
         ByteBuffer &operator >> (time_t &value)
         {
             value = read<time_t>();
diff --git a/src/shared/CMakeLists.txt b/src/shared/CMakeLists.txt
index 719a1c5..dfad818 100644
--- a/src/shared/CMakeLists.txt
+++ b/src/shared/CMakeLists.txt
@@ -154,7 +154,9 @@ if(WIN32)
     Turtle.natvis
     )
 
-  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /D__ACE_INLINE__")
+  if(MSVC)
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /D__ACE_INLINE__")
+  endif()
 else()
   list(REMOVE_ITEM shared_SRCS
     WheatyExceptionReport.cpp
diff --git a/src/shared/Common.h b/src/shared/Common.h
index 8c81db9..e630fcb 100644
--- a/src/shared/Common.h
+++ b/src/shared/Common.h
@@ -69,6 +69,7 @@
 #include <string.h>
 #include <time.h>
 #include <math.h>
+#include <cmath>
 #include <errno.h>
 #include <signal.h>
 #include <assert.h>
@@ -148,7 +149,7 @@ typedef off_t ACE_OFF_T;
 
 #define SIZEFMTD ACE_SIZE_T_FORMAT_SPECIFIER
 
-inline float finiteAlways(float f) { return finite(f) ? f : 0.0f; }
+inline float finiteAlways(float f) { return std::isfinite(f) ? f : 0.0f; }
 
 #define atol(a) strtoul( a, nullptr, 10)
 
