diff --git a/src/game/Chat/Channel.h b/src/game/Chat/Channel.h
index fa07485..96a5c27 100644
--- a/src/game/Chat/Channel.h
+++ b/src/game/Chat/Channel.h
@@ -318,7 +318,7 @@ class Channel
 
         typedef     std::map<ObjectGuid, PlayerInfo> PlayerList;
         PlayerList  m_players;
-        typedef     std::set<ObjectGuid> BannedList;
+        typedef     std::unordered_set<ObjectGuid> BannedList;
         BannedList  m_banned;
 };
 #endif
