diff --git a/src/game/ObjectMgr.cpp b/src/game/ObjectMgr.cpp
index 678b537..e85ec15 100644
--- a/src/game/ObjectMgr.cpp
+++ b/src/game/ObjectMgr.cpp
@@ -2199,7 +2199,7 @@ void ObjectMgr::CorrectItemEffects(uint32 itemId, _ItemSpell& itemSpell)
         itemSpell.SpellId = 0;
 }
 
-void ObjectMgr::FillObtainedItemsList(std::set<uint32>& obtainedItems)
+void ObjectMgr::FillObtainedItemsList(std::unordered_set<uint32>& obtainedItems)
 {
     // These are all the items that have been obtained by players.
     {
@@ -2348,7 +2348,7 @@ void ObjectMgr::LoadItemPrototypes()
     m_QuestStartingItemsMap.clear();
     
     // Load all currently obtained items by players.
-    std::set<uint32> obtainedItems;
+    std::unordered_set<uint32> obtainedItems;
     FillObtainedItemsList(obtainedItems);
 
     // check data correctness
diff --git a/src/game/ObjectMgr.h b/src/game/ObjectMgr.h
index 8c9325e..a0942b1 100644
--- a/src/game/ObjectMgr.h
+++ b/src/game/ObjectMgr.h
@@ -982,7 +982,7 @@ class ObjectMgr
         void LoadGameObjectLocales();
         void LoadGameobjects(bool reload = false);
         void LoadItemPrototypes();
-        void FillObtainedItemsList(std::set<uint32>&);
+        void FillObtainedItemsList(std::unordered_set<uint32>&);
         void CorrectItemEffects(uint32, _ItemSpell&);
         void LoadItemRequiredTarget();
         void LoadItemLocales();
