diff --git a/src/game/ObjectMgr.h b/src/game/ObjectMgr.h
index 06a0bc6..b144901 100644
--- a/src/game/ObjectMgr.h
+++ b/src/game/ObjectMgr.h
@@ -1724,16 +1724,16 @@ class ObjectMgr
         void LoadTrainers(char const* tableName, bool isTemplates);
 
         // Storing all existing IDs in database.
-        std::set<uint32> m_ItemIdSet;
-        std::set<uint32> m_QuestIdSet;
-        std::set<uint32> m_CreatureIdSet;
-        std::set<uint32> m_GameObjectIdSet;
-        std::set<uint32> m_CreatureGuidSet;
-        std::set<uint32> m_GameObjectGuidSet;
-        std::set<uint32> m_AreaTriggerIdSet;
-        std::set<uint32> m_CreatureSpellsIdSet;
-        std::set<uint32> m_VendorTemplateIdSet;
-        std::set<uint32> m_GossipMenuIdSet;
+        std::unordered_set<uint32> m_ItemIdSet;
+        std::unordered_set<uint32> m_QuestIdSet;
+        std::unordered_set<uint32> m_CreatureIdSet;
+        std::unordered_set<uint32> m_GameObjectIdSet;
+        std::unordered_set<uint32> m_CreatureGuidSet;
+        std::unordered_set<uint32> m_GameObjectGuidSet;
+        std::unordered_set<uint32> m_AreaTriggerIdSet;
+        std::unordered_set<uint32> m_CreatureSpellsIdSet;
+        std::unordered_set<uint32> m_VendorTemplateIdSet;
+        std::unordered_set<uint32> m_GossipMenuIdSet;
 
         typedef std::map<uint32,PetLevelInfo*> PetLevelInfoMap;
         // PetLevelInfoMap[creature_id][level]
