diff --git a/src/game/Objects/GameObject.h b/src/game/Objects/GameObject.h
index b056488..c31c503 100644
--- a/src/game/Objects/GameObject.h
+++ b/src/game/Objects/GameObject.h
@@ -895,14 +895,14 @@ class GameObject : public WorldObject
 
         typedef std::set<ObjectGuid> GuidsSet;
 
-        GuidsSet m_SkillupSet;                              // players that already have skill-up at GO use
+        ObjectGuidSet m_SkillupSet;                         // players that already have skill-up at GO use
 
         uint32 m_useTimes;                                  // amount uses/charges triggered
 
         // collected only for GAMEOBJECT_TYPE_SUMMONING_RITUAL
         ObjectGuid m_firstUser;                             // first GO user, in most used cases owner, but in some cases no, for example non-summoned multi-use GAMEOBJECT_TYPE_SUMMONING_RITUAL
         GuidsSet m_UniqueUsers;                             // all players who use item, some items activated after specific amount unique uses
-        GuidsSet m_allowedLooters;                          // if not empty only those players are allowed to use the gameobject
+        ObjectGuidSet m_allowedLooters;                     // if not empty only those players are allowed to use the gameobject
         std::shared_mutex m_UniqueUsers_lock;
         ObjectGuid m_summonTarget;                          // The player who is being summoned
 
