diff --git a/src/game/HardcodedEvents.cpp b/src/game/HardcodedEvents.cpp
index 9485e56..4de5810 100644
--- a/src/game/HardcodedEvents.cpp
+++ b/src/game/HardcodedEvents.cpp
@@ -807,18 +807,18 @@ void ScourgeInvasionEvent::Disable()
             Creature* pRelay = pMap->GetCreature(necro.relayGuid);
             if (!pRelay)
                 continue;
-            std::list<Creature*> shardList;
+            std::vector<Creature*> shardList;
             GetCreatureListWithEntryInGrid(shardList, pRelay, { NPC_NECROTIC_SHARD, NPC_DAMAGED_NECROTIC_SHARD }, 400.0f);
             for (Creature* pShard : shardList)
                 pShard->DeleteLater();
-            std::list<GameObject*> necropolisList;
+            std::vector<GameObject*> necropolisList;
             GetGameObjectListWithEntryInGrid(necropolisList, pRelay, GOBJ_NECROPOLIS, 100.0f);
             for (GameObject* pNecro : necropolisList)
                 pNecro->DeleteLater();
             
             // Getting list of relays as well, in case there's been some double enable/disabling going on 
             // and we have more than one relay alive
-            std::list<Creature*> relayList;
+            std::vector<Creature*> relayList;
             GetCreatureListWithEntryInGrid(relayList, pRelay, NPC_NECROPOLIS_RELAY, 100.0f);
             for (Creature* p2Relay : relayList)
                 p2Relay->DeleteLater();
diff --git a/src/game/Maps/GridSearchers.h b/src/game/Maps/GridSearchers.h
index 84f6e60..653b424 100644
--- a/src/game/Maps/GridSearchers.h
+++ b/src/game/Maps/GridSearchers.h
@@ -44,6 +44,33 @@ void GetGameObjectListWithEntryInGrid(std::list<GameObject*>& lList , WorldObjec
 void GetCreatureListWithEntryInGrid(std::list<Creature*>& lList, WorldObject* pSource, uint32 uiEntry, float fMaxSearchRange);
 void GetCreatureListWithEntryInGrid(std::list<Creature*>& lList, WorldObject* pSource, const std::vector<uint32>& entries, float fMaxSearchRange);
 
+template<class Container>
+void GetGameObjectListWithEntryInGrid(Container& objects, WorldObject* source, uint32 entry, float maxSearchRange)
+{
+    ASSERT(source);
+    MaNGOS::AllGameObjectsWithEntryInRange check(source, entry, maxSearchRange);
+    MaNGOS::GameObjectListSearcher<MaNGOS::AllGameObjectsWithEntryInRange, Container> searcher(objects, check);
+    Cell::VisitGridObjects(source, searcher, maxSearchRange);
+}
+
+template<class Container>
+void GetCreatureListWithEntryInGrid(Container& creatures, WorldObject* source, uint32 entry, float maxSearchRange)
+{
+    ASSERT(source);
+    MaNGOS::AllCreaturesOfEntryInRange check(source, entry, maxSearchRange);
+    MaNGOS::CreatureListSearcher<MaNGOS::AllCreaturesOfEntryInRange, Container> searcher(creatures, check);
+    Cell::VisitGridObjects(source, searcher, maxSearchRange);
+}
+
+template<class Container>
+void GetCreatureListWithEntryInGrid(Container& creatures, WorldObject* source, const std::vector<uint32>& entries, float maxSearchRange)
+{
+    ASSERT(source);
+    MaNGOS::AllCreaturesMatchingOneEntryInRange check(source, entries, maxSearchRange);
+    MaNGOS::CreatureListSearcher<MaNGOS::AllCreaturesMatchingOneEntryInRange, Container> searcher(creatures, check);
+    Cell::VisitGridObjects(source, searcher, maxSearchRange);
+}
+
 void GetHostileCreaturesListInRange(std::list<Creature*>& lList, WorldObject* pSource, float fMaxSearchRange);
 
 /*
diff --git a/src/game/Maps/ScriptCommands.cpp b/src/game/Maps/ScriptCommands.cpp
index e7d1681..f335879 100644
--- a/src/game/Maps/ScriptCommands.cpp
+++ b/src/game/Maps/ScriptCommands.cpp
@@ -420,7 +420,7 @@ bool Map::ScriptCommand_SummonCreature(ScriptInfo const& script, WorldObject* so
     if (script.summonCreature.flags & (SF_SUMMONCREATURE_UNIQUE | SF_SUMMONCREATURE_UNIQUE_TEMP))
     {
         float dist = script.summonCreature.uniqueDistance ? script.summonCreature.uniqueDistance : (pSummoner->GetDistance(x, y, z) + 50.0f) * 2;
-        std::list<Creature*> foundCreatures;
+        std::vector<Creature*> foundCreatures;
 
         GetCreatureListWithEntryInGrid(foundCreatures, pSummoner, script.summonCreature.creatureEntry, dist);
 
