diff --git a/src/game/Anticheat/WardenAnticheat/WardenMgr.cpp b/src/game/Anticheat/WardenAnticheat/WardenMgr.cpp
index 61e4661..80a30ef 100644
--- a/src/game/Anticheat/WardenAnticheat/WardenMgr.cpp
+++ b/src/game/Anticheat/WardenAnticheat/WardenMgr.cpp
@@ -89,11 +89,10 @@ void WardenMgr::LoadWardenChecks()
 
             if (wardenCheck->Data.GetNumBytes() < len)
             {
-                uint8 temp[24];
-                memset(temp, 0, len);
-                memcpy(temp, wardenCheck->Data.AsByteArray().data(), wardenCheck->Data.GetNumBytes());
-                std::reverse(temp, temp + len);
-                wardenCheck->Data.SetBinary((uint8*)temp, len);
+                std::vector<uint8> temp(len, 0);
+                memcpy(temp.data(), wardenCheck->Data.AsByteArray().data(), wardenCheck->Data.GetNumBytes());
+                std::reverse(temp.begin(), temp.end());
+                wardenCheck->Data.SetBinary(temp.data(), len);
             }
         }
 
@@ -117,12 +116,10 @@ void WardenMgr::LoadWardenChecks()
             int len = checkResult.size() / 2;
             if (wr->Result.GetNumBytes() < len)
             {
-                uint8 *temp = new uint8[len];
-                memset(temp, 0, len);
-                memcpy(temp, wr->Result.AsByteArray().data(), wr->Result.GetNumBytes());
-                std::reverse(temp, temp + len);
-                wr->Result.SetBinary((uint8*)temp, len);
-                delete[] temp;
+                std::vector<uint8> temp(len, 0);
+                memcpy(temp.data(), wr->Result.AsByteArray().data(), wr->Result.GetNumBytes());
+                std::reverse(temp.begin(), temp.end());
+                wr->Result.SetBinary(temp.data(), len);
             }
 
             CheckResultStore.insert(std::pair<uint16, WardenCheckResult*>(build, wr));
diff --git a/src/game/Anticheat/WardenAnticheat/WardenWin.cpp b/src/game/Anticheat/WardenAnticheat/WardenWin.cpp
index 264d8ea..d4a88e5 100644
--- a/src/game/Anticheat/WardenAnticheat/WardenWin.cpp
+++ b/src/game/Anticheat/WardenAnticheat/WardenWin.cpp
@@ -413,13 +413,7 @@ void WardenWin::HandleData(ByteBuffer &buff)
                 buff >> luaStrLen;
 
                 if (luaStrLen != 0)
-                {
-                    char *str = new char[luaStrLen + 1];
-                    memcpy(str, buff.contents() + buff.rpos(), luaStrLen);
-                    str[luaStrLen] = '\0'; // Null terminator
-                    sLog.outWardenDebug("Lua string: %s", str);
-                    delete[] str;
-                }
+                    sLog.outWardenDebug("Lua string: %.*s", int(luaStrLen), reinterpret_cast<char const*>(buff.contents() + buff.rpos()));
                 buff.rpos(buff.rpos() + luaStrLen); // Skip string
                 sLog.outWardenDebug("RESULT LUA_STR_CHECK passed, CheckId %u account Id %u", *itr, m_session->GetAccountId());
                 break;
