diff --git a/src/game/Objects/Player.cpp b/src/game/Objects/Player.cpp
index 61514ee..ed01c4f 100644
--- a/src/game/Objects/Player.cpp
+++ b/src/game/Objects/Player.cpp
@@ -4578,7 +4578,8 @@ void Player::LearnSpell(uint32 spell_id, bool dependent, bool talent)
     if (disabled || talent)
     {
         SpellChainMapNext const& nextMap = sSpellMgr.GetSpellChainNext();
-        for (SpellChainMapNext::const_iterator i = nextMap.lower_bound(spell_id); i != nextMap.upper_bound(spell_id); ++i)
+        auto const range = nextMap.equal_range(spell_id);
+        for (auto i = range.first; i != range.second; ++i)
         {
             PlayerSpellMap::iterator iter = m_spells.find(i->second);
             if (iter != m_spells.end() && iter->second.disabled)
@@ -4612,7 +4613,8 @@ void Player::RemoveSpell(uint32 spell_id, bool disabled, bool learn_low_rank, bo
 
                           // unlearn non talent higher ranks (recursive)
     SpellChainMapNext const& nextMap = sSpellMgr.GetSpellChainNext();
-    for (SpellChainMapNext::const_iterator itr2 = nextMap.lower_bound(spell_id); itr2 != nextMap.upper_bound(spell_id); ++itr2)
+    auto const range = nextMap.equal_range(spell_id);
+    for (auto itr2 = range.first; itr2 != range.second; ++itr2)
         if (HasSpell(itr2->second) && !GetTalentSpellPos(itr2->second))
             RemoveSpell(itr2->second, !Spells::IsPassiveSpell(itr2->second), false);
 
diff --git a/src/game/Spells/SpellMgr.h b/src/game/Spells/SpellMgr.h
index cdab8f8..9293631 100644
--- a/src/game/Spells/SpellMgr.h
+++ b/src/game/Spells/SpellMgr.h
@@ -526,7 +526,8 @@ class SpellMgr
         void doForHighRanks(uint32 spellid, Worker& worker)
         {
             SpellChainMapNext const& nextMap = GetSpellChainNext();
-            for(SpellChainMapNext::const_iterator itr = nextMap.lower_bound(spellid); itr != nextMap.upper_bound(spellid); ++itr)
+            auto const range = nextMap.equal_range(spellid);
+            for (auto itr = range.first; itr != range.second; ++itr)
             {
                 worker(itr->second);
                 doForHighRanks(itr->second,worker);
