diff --git a/src/game/Guild/Guild.cpp b/src/game/Guild/Guild.cpp
index 49bb997..32873f1 100644
--- a/src/game/Guild/Guild.cpp
+++ b/src/game/Guild/Guild.cpp
@@ -104,7 +104,7 @@ bool Guild::Create(Petition* petition, Player* leader)
     if (!Create(leader, petition->GetName()))
         return false;
 
-    PetitionSignatureList signatures = petition->GetSignatureList();
+    PetitionSignatureList const& signatures = petition->GetSignatureList();
     for (auto iter = signatures.cbegin(); iter != signatures.cend(); ++iter)
     {
         PetitionSignature* signature = *iter;
