diff --git a/src/shared/Common.h b/src/shared/Common.h
index e630fcb..a3d7935 100644
--- a/src/shared/Common.h
+++ b/src/shared/Common.h
@@ -255,8 +255,9 @@ extern LocaleNameStr const fullLocaleNameList[];
 //operator new[] based version of strdup() function! Release memory by using operator delete[] !
 inline char * mangos_strdup(const char * source)
 {
-    char * dest = new char[strlen(source) + 1];
-    strcpy(dest, source);
+    size_t const size = strlen(source) + 1;
+    char * dest = new char[size];
+    memcpy(dest, source, size);
     return dest;
 }
 
