diff --git a/src/game/LFG/LFGMgr.cpp b/src/game/LFG/LFGMgr.cpp
index a55c88d..d257e12 100644
--- a/src/game/LFG/LFGMgr.cpp
+++ b/src/game/LFG/LFGMgr.cpp
@@ -52,6 +52,9 @@ void LFGPlayerQueueInfo::CalculateRoles(Classes playerClass)
 {
     roleMask = LFGQueue::CalculateRoles(playerClass);
 
+    if (rolePriority.empty())
+        rolePriority.reserve(PotentialRoles.size());
+
     // Determine role priority
     for (ClassRoles role : PotentialRoles)
     {
diff --git a/src/game/LFG/LFGMgr.h b/src/game/LFG/LFGMgr.h
index 88a0ae9..2e3542a 100644
--- a/src/game/LFG/LFGMgr.h
+++ b/src/game/LFG/LFGMgr.h
@@ -25,7 +25,7 @@
 #ifndef MANGOSSERVER_LFGMGR_H
 #define MANGOSSERVER_LFGMGR_H
 
-#include <list>
+#include <vector>
 #include <map>
 
 #include "Policies/Singleton.h"
@@ -70,7 +70,7 @@ struct LFGPlayerQueueInfo
     bool hasQueuePriority;
     bool isHardcore = false;
     std::string name;
-    std::list<std::pair<ClassRoles, RolesPriority>> rolePriority;
+    std::vector<std::pair<ClassRoles, RolesPriority>> rolePriority;
 
     void CalculateRoles(Classes playerClass);
     RolesPriority GetRolePriority(ClassRoles role);
