diff --git a/modules/mod-playerbots/src/playerbot/strategy/actions/AddLootAction.cpp b/modules/mod-playerbots/src/playerbot/strategy/actions/AddLootAction.cpp
index f6033d1984736851aaf000445f58c92955b2d95c..f9a357a1d86d343701ba4d03c0fd522f9e83ca1e 100644
--- a/modules/mod-playerbots/src/playerbot/strategy/actions/AddLootAction.cpp
+++ b/modules/mod-playerbots/src/playerbot/strategy/actions/AddLootAction.cpp
@@ -120,8 +120,8 @@ bool AddAllLootAction::AddLoot(Player* requester, ObjectGuid guid)
         //NOTE: They are !unable to loot quests items! too if so
         if (isInDungeon
             && group->GetLootMethod() == LootMethod::MASTER_LOOT
-            && group->GetMasterLooterGuid()
-            && group->GetMasterLooterGuid() != bot->GetObjectGuid())
+            && group->GetLooterGuid()
+            && group->GetLooterGuid() != bot->GetObjectGuid())
         {
             ai->TellDebug(requester, "Not master looter.", "debug loot");
             return false;
