diff --git a/modules/mod-playerbots/src/playerbot/strategy/actions/MailAction.cpp b/modules/mod-playerbots/src/playerbot/strategy/actions/MailAction.cpp
index d4843817f9e8c188aa8070d27bb6cc96fe053420..9d392454217b03a14c52cac7357a878f5ec7c079 100644
--- a/modules/mod-playerbots/src/playerbot/strategy/actions/MailAction.cpp
+++ b/modules/mod-playerbots/src/playerbot/strategy/actions/MailAction.cpp
@@ -1,5 +1,6 @@
 
 #include "Mail/Mail.h"
+#include "MapNodes/MasterPlayer.h"
 #include "playerbot/playerbot.h"
 #include "MailAction.h"
 #include "playerbot/PlayerbotAIConfig.h"
@@ -22,6 +23,7 @@ public:
     bool Process(Player* requester, int index, Mail* mail, PlayerbotAI* ai, Event& event) override
     {
         Player* bot = ai->GetBot();
+        MasterPlayer* masterPlayer = bot->GetSession()->GetMasterPlayer();
         time_t cur_time = time(0);
         int days = (cur_time - mail->deliver_time) / 3600 / 24;
         std::ostringstream out;
@@ -39,7 +41,7 @@ public:
         {
             for (MailItemInfoVec::iterator i = mail->items.begin(); i != mail->items.end(); ++i)
             {
-                Item* item = bot->GetMItem(i->item_guid);
+                Item* item = masterPlayer ? masterPlayer->GetMItem(i->item_guid) : nullptr;
                 int count = item ? item->GetCount() : 1;
                 ItemPrototype const *proto = sObjectMgr.GetItemPrototype(i->item_template);
                 if (proto)
@@ -86,6 +88,10 @@ public:
     bool Process(Player* requester, int index, Mail* mail, PlayerbotAI* ai, Event& event) override
     {
         Player* bot = ai->GetBot();
+        MasterPlayer* masterPlayer = bot->GetSession()->GetMasterPlayer();
+        if (!masterPlayer)
+            return false;
+
         if (!CheckBagSpace(bot))
         {
             ai->TellError(requester, "Not enough bag space");
@@ -130,7 +136,7 @@ public:
 #ifndef MANGOSBOT_ZERO
                 packet << *i;
 #endif
-                Item* item = bot->GetMItem(*i);
+                Item* item = masterPlayer->GetMItem(*i);
 
                 if (item)
                 {
@@ -303,9 +309,13 @@ bool MailAction::Execute(Event& event)
     if (!processor->Before(requester, ai))
         return false;
 
+    MasterPlayer* masterPlayer = bot->GetSession()->GetMasterPlayer();
+    if (!masterPlayer)
+        return false;
+
     std::vector<Mail*> mailList;
     time_t cur_time = time(0);
-    for (PlayerMails::iterator itr = bot->GetMailBegin(); itr != bot->GetMailEnd(); ++itr)
+    for (PlayerMails::iterator itr = masterPlayer->GetMailBegin(); itr != masterPlayer->GetMailEnd(); ++itr)
     {
         if ((*itr)->state == MAIL_STATE_DELETED || cur_time < (*itr)->deliver_time)
             continue;
