diff --git a/modules/mod-playerbots/src/playerbot/strategy/actions/MoveToRpgTargetAction.cpp b/modules/mod-playerbots/src/playerbot/strategy/actions/MoveToRpgTargetAction.cpp
index 9c96b12713d03145084fa1195a0838076ad4ecd3..1fbc5f770d76fe9fc10ea01a3129b16361bb97b9 100644
--- a/modules/mod-playerbots/src/playerbot/strategy/actions/MoveToRpgTargetAction.cpp
+++ b/modules/mod-playerbots/src/playerbot/strategy/actions/MoveToRpgTargetAction.cpp
@@ -171,17 +171,14 @@ bool MoveToRpgTargetAction::Execute(Event& event)
 
     bool couldMove;
 
-    if (unit && unit->IsMoving() && bot->GetDistance(unit) < INTERACTION_DISTANCE * 2 && unit->GetMotionMaster()->GetCurrentMovementGeneratorType() != IDLE_MOTION_TYPE)
+    if (unit && unit->GetTypeId() == TYPEID_UNIT && unit->IsMoving() &&
+        bot->GetDistance(unit) < INTERACTION_DISTANCE * 2 &&
+        unit->GetMotionMaster()->GetCurrentMovementGeneratorType() != IDLE_MOTION_TYPE)
     {
-
-        Creature* creature = static_cast<Creature*>(unit);
-
-
-        if (creature)
-            if (uint32 pauseTimer = creature->GetInteractionPauseTimer())
-                creature->GetMotionMaster()->PauseWaypoints(pauseTimer);
+        static_cast<Creature*>(unit)->PauseOutOfCombatMovement();
     }
-        couldMove = MoveTo(mapId, x, y, z, false, false);
+
+    couldMove = MoveTo(mapId, x, y, z, false, false);
 
     if (!couldMove && movePos.distance(bot) > INTERACTION_DISTANCE)
     {
