diff --git a/modules/mod-playerbots/src/playerbot/strategy/actions/SendMailAction.cpp b/modules/mod-playerbots/src/playerbot/strategy/actions/SendMailAction.cpp
index 61babceb06e04e6b199d7326695a5badf5315a5f..8e182023bf92c9018be1b0d7d859aa96533db8fc 100644
--- a/modules/mod-playerbots/src/playerbot/strategy/actions/SendMailAction.cpp
+++ b/modules/mod-playerbots/src/playerbot/strategy/actions/SendMailAction.cpp
@@ -33,10 +33,21 @@ bool SendMailAction::Execute(Event& event)
 
     if (!tellTo) tellTo = receiver;
 
+    auto whisperToTell = [this, tellTo](std::string const& message)
+    {
+        if (!tellTo->GetSession())
+            return;
+
+        WorldPacket data;
+        ChatHandler::BuildChatPacket(data, CHAT_MSG_WHISPER, message, LANG_UNIVERSAL, bot->GetChatTag(),
+            bot->GetObjectGuid(), bot->GetName(), tellTo->GetObjectGuid(), tellTo->GetName());
+        tellTo->GetSession()->SendPacket(&data);
+    };
+
     ItemIds ids = chat->parseItems(text);
     if (ids.size() > 1)
     {
-        bot->Whisper("You can not request more than one item", LANG_UNIVERSAL, tellTo->GetObjectGuid());
+        whisperToTell("You can not request more than one item");
         return false;
     }
 
@@ -48,7 +59,7 @@ bool SendMailAction::Execute(Event& event)
 
         if (randomBot)
         {
-            bot->Whisper("I cannot send money", LANG_UNIVERSAL, tellTo->GetObjectGuid());
+            whisperToTell("I cannot send money");
             return false;
         }
 
@@ -99,7 +110,7 @@ bool SendMailAction::Execute(Event& event)
             {
                 std::ostringstream out;
                 out << "Cannot send " << ChatHelper::formatItem(item);
-                bot->Whisper(out.str(), LANG_UNIVERSAL, tellTo->GetObjectGuid());
+                whisperToTell(out.str());
                 continue;
             }
 
@@ -116,7 +127,7 @@ bool SendMailAction::Execute(Event& event)
                 {
                     std::ostringstream out;
                     out << ChatHelper::formatItem(item) << ": it is not for sale";
-                    bot->Whisper(out.str(), LANG_UNIVERSAL, tellTo->GetObjectGuid());
+                    whisperToTell(out.str());
                     return false;
                 }
                 draft.SetCOD(price);
@@ -124,7 +135,7 @@ bool SendMailAction::Execute(Event& event)
             draft.SendMailTo(MailReceiver(receiver), MailSender(bot));
 
             std::ostringstream out; out << "Sent mail to " << receiver->GetName();
-            bot->Whisper(out.str(), LANG_UNIVERSAL, tellTo->GetObjectGuid());
+            whisperToTell(out.str());
             return true;
         }
     }
