diff --git a/modules/mod-playerbots/src/playerbot/strategy/actions/WtsAction.cpp b/modules/mod-playerbots/src/playerbot/strategy/actions/WtsAction.cpp
index a37a15f9c2a12e8c0940353e1111b0fea52888a6..74e170e06c01f7d92d51260594a6ea192b119735 100644
--- a/modules/mod-playerbots/src/playerbot/strategy/actions/WtsAction.cpp
+++ b/modules/mod-playerbots/src/playerbot/strategy/actions/WtsAction.cpp
@@ -49,7 +49,13 @@ bool WtsAction::Execute(Event& event)
         tell << "I'll buy " << chat->formatItem(proto) << " for " << chat->formatMoney(buyPrice);
 
         // ignore random bot chat filter
-        bot->Whisper(tell.str(), LANG_UNIVERSAL, owner->GetObjectGuid());
+        if (owner->GetSession())
+        {
+            WorldPacket data;
+            ChatHandler::BuildChatPacket(data, CHAT_MSG_WHISPER, tell.str(), LANG_UNIVERSAL, bot->GetChatTag(),
+                bot->GetObjectGuid(), bot->GetName(), owner->GetObjectGuid(), owner->GetName());
+            owner->GetSession()->SendPacket(&data);
+        }
     }
 
     return true;
