diff --git a/src/game/Movement/spline/MoveSpline.cpp b/src/game/Movement/spline/MoveSpline.cpp
index 07c6fb5..bd28008 100644
--- a/src/game/Movement/spline/MoveSpline.cpp
+++ b/src/game/Movement/spline/MoveSpline.cpp
@@ -242,20 +242,20 @@ MoveSpline::UpdateResult MoveSpline::_updateState(int32& ms_time_diff)
 std::string MoveSpline::ToString() const
 {
     std::stringstream str;
-    str << "MoveSpline" << std::endl;
-    str << "spline Id: " << GetId() << std::endl;
-    str << "flags: " << splineflags.ToString() << std::endl;
+    str << "MoveSpline" << '\n';
+    str << "spline Id: " << GetId() << '\n';
+    str << "flags: " << splineflags.ToString() << '\n';
     if (splineflags.final_angle)
         str << "facing  angle: " << facing.angle;
     else if (splineflags.final_target)
         str << "facing target: " << facing.target;
     else if (splineflags.final_point)
         str << "facing  point: " << facing.f.x << " " << facing.f.y << " " << facing.f.z;
-    str << std::endl;
-    str << "time passed: " << time_passed << std::endl;
-    str << "total  time: " << Duration() << std::endl;
-    str << "spline point Id: " << point_Idx << std::endl;
-    str << "path  point  Id: " << currentPathIdx() << std::endl;
+    str << '\n';
+    str << "time passed: " << time_passed << '\n';
+    str << "total  time: " << Duration() << '\n';
+    str << "spline point Id: " << point_Idx << '\n';
+    str << "path  point  Id: " << currentPathIdx() << '\n';
     str << spline.ToString();
     return str.str();
 }
diff --git a/src/game/Movement/spline/spline.cpp b/src/game/Movement/spline/spline.cpp
index 3fe2345..58eb2af 100644
--- a/src/game/Movement/spline/spline.cpp
+++ b/src/game/Movement/spline/spline.cpp
@@ -296,10 +296,10 @@ std::string SplineBase::ToString() const
     const char* mode_str[ModesEnd] = {"Linear", "CatmullRom", "Bezier3", "Uninitialized"};
 
     index_type count = this->points.size();
-    str << "mode: " << mode_str[mode()] << std::endl;
-    str << "points count: " << count << std::endl;
+    str << "mode: " << mode_str[mode()] << '\n';
+    str << "points count: " << count << '\n';
     for (index_type i = 0; i < count; ++i)
-        str << "point " << i << " : " << points[i].toString() << std::endl;
+        str << "point " << i << " : " << points[i].toString() << '\n';
 
     return str.str();
 }
