diff --git a/src/game/Mail/Mail.cpp b/src/game/Mail/Mail.cpp
index dd9b80e..12b54b8 100644
--- a/src/game/Mail/Mail.cpp
+++ b/src/game/Mail/Mail.cpp
@@ -126,18 +126,18 @@ MailDraft::MailDraft(uint16 mailTemplateId, bool need_items, LocaleConstant loca
  * @param subject The subject of the mail.
  * @param itemText The text of the body of the mail.
  */
-MailDraft::MailDraft(std::string subject, std::string text) : m_mailTemplateId(0), m_mailTemplateItemsNeed(false), m_subject(subject),
-    m_bodyId(!text.empty() ? sObjectMgr.CreateItemText(text) : 0), m_money(0), m_COD(0)
+MailDraft::MailDraft(std::string subject, std::string text) : m_mailTemplateId(0), m_mailTemplateItemsNeed(false), m_subject(std::move(subject)),
+    m_bodyId(!text.empty() ? sObjectMgr.CreateItemText(std::move(text)) : 0), m_money(0), m_COD(0)
 {
 
 }
 
 MailDraft& MailDraft::SetSubjectAndBody(std::string subject, std::string text)
 {
-    m_subject = subject;
+    m_subject = std::move(subject);
 
     MANGOS_ASSERT(!m_bodyId);
-    m_bodyId = !text.empty() ? sObjectMgr.CreateItemText(text) : 0;
+    m_bodyId = !text.empty() ? sObjectMgr.CreateItemText(std::move(text)) : 0;
 
     return *this;
 }
diff --git a/src/game/Mail/Mail.h b/src/game/Mail/Mail.h
index f8bf8d6..2e7f431 100644
--- a/src/game/Mail/Mail.h
+++ b/src/game/Mail/Mail.h
@@ -40,6 +40,7 @@
 #include "Common.h"
 #include "ObjectGuid.h"
 #include <map>
+#include <utility>
 
 struct AuctionEntry;
 class Item;
@@ -209,7 +210,7 @@ class MailDraft
          * @param itemTextId The id of the body of the mail.
          */
         MailDraft(std::string subject, uint32 itemTextId = 0)
-            : m_mailTemplateId(0), m_mailTemplateItemsNeed(false), m_subject(subject), m_bodyId(itemTextId), m_money(0), m_COD(0) {}
+            : m_mailTemplateId(0), m_mailTemplateItemsNeed(false), m_subject(std::move(subject)), m_bodyId(itemTextId), m_money(0), m_COD(0) {}
         /**
          * Creates a new MailDraft object using subject and contect texts.
          *
@@ -231,7 +232,7 @@ class MailDraft
     public:                                                 // modifiers
 
         // this two modifiers expected to be applied in normal case to blank draft and exclusively, It DON'T must overwrite already set itemTextId, in other cases it will work and with mixed cases but this will be not normal way use.
-        MailDraft& SetSubjectAndBodyId(std::string subject, uint32 itemTextId) { m_subject = subject; MANGOS_ASSERT(!m_bodyId); m_bodyId = itemTextId; return *this; }
+        MailDraft& SetSubjectAndBodyId(std::string subject, uint32 itemTextId) { m_subject = std::move(subject); MANGOS_ASSERT(!m_bodyId); m_bodyId = itemTextId; return *this; }
         MailDraft& SetSubjectAndBody(std::string subject, std::string text);
         MailDraft& SetMailTemplate(uint16 mailTemplateId, bool need_items = true) { m_mailTemplateId = mailTemplateId, m_mailTemplateItemsNeed = need_items; return *this; }
 
