diff --git a/modules/mod-playerbots/src/playerbot/PlayerbotLoginMgr.cpp b/modules/mod-playerbots/src/playerbot/PlayerbotLoginMgr.cpp
index 353b453..d722e51 100644
--- a/modules/mod-playerbots/src/playerbot/PlayerbotLoginMgr.cpp
+++ b/modules/mod-playerbots/src/playerbot/PlayerbotLoginMgr.cpp
@@ -4,6 +4,7 @@
 #include "PlayerbotMgr.h"
 #include "PlayerbotAIConfig.h"
 #include "RandomPlayerbotMgr.h"
+#include "Timer.h"
 
 using namespace ai;
 
@@ -28,7 +29,7 @@ public:
 
 PlayerLoginInfo::PlayerLoginInfo(const uint32 account, const uint32 guid, const uint8 race, const uint8 cls, const uint32 level, const bool isNew, const WorldPosition& position, const uint32 guildId) : account(account), guid(guid), race(race), cls(cls), level(level), isNew(isNew), position(position), guildId(guildId) {}
 
-PlayerLoginInfo::PlayerLoginInfo(Player* player) : PlayerLoginInfo(player->GetSession()->GetAccountId(), player->GetDbGuid(), player->GetRace(), player->GetClass(), player->GetLevel(), player->GetTotalPlayedTime() == 0, player, player->GetGuildId()) {};
+PlayerLoginInfo::PlayerLoginInfo(Player* player) : PlayerLoginInfo(player->GetSession()->GetAccountId(), player->GetGUIDLow(), player->GetRace(), player->GetClass(), player->GetLevel(), player->GetTotalPlayedTime() == 0, player, player->GetGuildId()) {};
 
 uint32 PlayerLoginInfo::GetLevel() const
 {
@@ -55,7 +56,7 @@ bool PlayerLoginInfo::IsNearPlayer(const LoginSpace& space) const
     for (auto& player : space.realPlayerInfos)
     {
         WorldPosition p(player.position);
-        if (p.mapid == position.mapid && p.sqDistance(position) < sPlayerbotAIConfig.loginBotsNearPlayerRange * sPlayerbotAIConfig.loginBotsNearPlayerRange)
+        if (p.mapId == position.mapId && p.sqDistance(position) < sPlayerbotAIConfig.loginBotsNearPlayerRange * sPlayerbotAIConfig.loginBotsNearPlayerRange)
         {
             return true;
         }
@@ -72,7 +73,7 @@ bool PlayerLoginInfo::IsOnPlayerMap(const LoginSpace& space) const
     for (auto& player : space.realPlayerInfos)
     {
         WorldPosition p(player.position);
-        if (p.mapid == position.mapid)
+        if (p.mapId == position.mapId)
         {
             return true;
         }
@@ -286,7 +287,7 @@ bool PlayerLoginInfo::LoginBot()
     if (holderState != HolderState::HOLDER_RECEIVED)
         return false;
 
-    if (sObjectMgr.GetPlayer(ObjectGuid(HIGHGUID_PLAYER, guid), false))
+    if (sObjectMgr.GetPlayer(ObjectGuid(HIGHGUID_PLAYER, guid)))
     {
         loginState = LoginState::BOT_ONLINE;
         return false;
@@ -299,7 +300,7 @@ bool PlayerLoginInfo::LoginBot()
     if(sPlayerbotAIConfig.randomBotTimedLogout)
         sRandomPlayerbotMgr.SetValue(guid, "add", 1, "", urand(sPlayerbotAIConfig.minRandomBotInWorldTime, sPlayerbotAIConfig.maxRandomBotInWorldTime));
 
-    Player* player = sObjectMgr.GetPlayer(ObjectGuid(HIGHGUID_PLAYER, guid), false);
+    Player* player = sObjectMgr.GetPlayer(ObjectGuid(HIGHGUID_PLAYER, guid));
 
     if (!player)
     {
@@ -319,7 +320,7 @@ bool PlayerLoginInfo::LogoutBot()
     if (loginState != LoginState::BOT_ON_LOGOUTQUEUE)
         return false;
 
-    Player* player = sObjectMgr.GetPlayer(ObjectGuid(HIGHGUID_PLAYER, guid), false);
+    Player* player = sObjectMgr.GetPlayer(ObjectGuid(HIGHGUID_PLAYER, guid));
 
     if (!player)
     {
@@ -333,7 +334,7 @@ bool PlayerLoginInfo::LogoutBot()
 
     sRandomPlayerbotMgr.LogoutPlayerBot(guid);
 
-    if (sObjectMgr.GetPlayer(ObjectGuid(HIGHGUID_PLAYER, guid), false))
+    if (sObjectMgr.GetPlayer(ObjectGuid(HIGHGUID_PLAYER, guid)))
         return false;
 
     loginState = LoginState::BOT_OFFLINE;    
@@ -450,7 +451,7 @@ BotPool PlayerBotLoginMgr::LoadBotsFromDb()
 
 void PlayerBotLoginMgr::SendHolders(const BotInfos& queue)
 {  
-    CharacterDatabase.AsyncPQuery(&RandomPlayerbotMgr::DatabasePing, sWorld.GetCurrentMSTime(), std::string("CharacterDatabase"), "select 1");
+    CharacterDatabase.AsyncPQuery(&RandomPlayerbotMgr::DatabasePing, WorldTimer::getMSTime(), std::string("CharacterDatabase"), "select 1");
 
     for (auto& info : queue)
     {
@@ -462,7 +463,7 @@ void PlayerBotLoginMgr::SendHolders(const BotInfos& queue)
 
 void PlayerBotLoginMgr::SendHolders(BotPool* pool)
 {
-    CharacterDatabase.AsyncPQuery(&RandomPlayerbotMgr::DatabasePing, sWorld.GetCurrentMSTime(), std::string("CharacterDatabase"), "select 1");
+    CharacterDatabase.AsyncPQuery(&RandomPlayerbotMgr::DatabasePing, WorldTimer::getMSTime(), std::string("CharacterDatabase"), "select 1");
 
     for (auto& [guid, info] : *pool)
     {
diff --git a/modules/mod-playerbots/src/playerbot/PlayerbotLoginMgr.h b/modules/mod-playerbots/src/playerbot/PlayerbotLoginMgr.h
index fa396ba..f85b751 100644
--- a/modules/mod-playerbots/src/playerbot/PlayerbotLoginMgr.h
+++ b/modules/mod-playerbots/src/playerbot/PlayerbotLoginMgr.h
@@ -1,5 +1,7 @@
 #include "WorldPosition.h"
 
+#include <future>
+
 namespace ai
 {
 	typedef std::map<uint32, Player*> RealPlayers;
