diff --git a/src/game/AccountMgr.cpp b/src/game/AccountMgr.cpp
index 3b5c277..b76c8a0 100644
--- a/src/game/AccountMgr.cpp
+++ b/src/game/AccountMgr.cpp
@@ -213,9 +213,9 @@ void AccountMgr::LoadGmLevels()
         case SEC_DEVELOPER:
         case SEC_ADMINISTRATOR:
         case SEC_SIGMACHAD:
-            if (m_accountSecurity.find(accountId) == m_accountSecurity.end() ||
-                m_accountSecurity[accountId] < secu)
-                m_accountSecurity[accountId] = secu;
+            auto [securityItr, inserted] = m_accountSecurity.emplace(accountId, secu);
+            if (!inserted && securityItr->second < secu)
+                securityItr->second = secu;
             break;
         }
     } while (result->NextRow());
diff --git a/src/game/Battlegrounds/BattleGroundMgr.cpp b/src/game/Battlegrounds/BattleGroundMgr.cpp
index 3885fe6..8660ad6 100644
--- a/src/game/Battlegrounds/BattleGroundMgr.cpp
+++ b/src/game/Battlegrounds/BattleGroundMgr.cpp
@@ -1398,11 +1398,12 @@ void BattleGroundMgr::ReloadBGPlayerCounts()
     {
         for (auto& bg : bgSet)
         {
-            if (newPlayerCounts.find(bg.second->GetTypeID()) == newPlayerCounts.end())
+            auto countItr = newPlayerCounts.find(bg.second->GetTypeID());
+            if (countItr == newPlayerCounts.end())
                 continue;
 
-            bg.second->SetMinPlayersPerTeam(newPlayerCounts[bg.second->GetTypeID()].first);
-            bg.second->SetMaxPlayersPerTeam(newPlayerCounts[bg.second->GetTypeID()].second);
+            bg.second->SetMinPlayersPerTeam(countItr->second.first);
+            bg.second->SetMaxPlayersPerTeam(countItr->second.second);
         }
     }
 }
diff --git a/src/shared/Database/Database.cpp b/src/shared/Database/Database.cpp
index d2848ca..f3b7d3d 100644
--- a/src/shared/Database/Database.cpp
+++ b/src/shared/Database/Database.cpp
@@ -657,7 +657,7 @@ SqlStatement Database::CreateStatement(SqlStatementID& index, const char * fmt )
         if(iter == m_stmtRegistry.end())
         {
             nId = ++m_iStmtIndex;
-            m_stmtRegistry[szFmt] = nId;
+            m_stmtRegistry.emplace(std::move(szFmt), nId);
         }
         else
             nId = iter->second;
