diff --git a/src/game/ObjectMgr.cpp b/src/game/ObjectMgr.cpp
index 2832261..b51ed68 100644
--- a/src/game/ObjectMgr.cpp
+++ b/src/game/ObjectMgr.cpp
@@ -4231,16 +4231,15 @@ void ObjectMgr::LoadQuests()
                 qinfo->RewMailTemplateId = 0;               // no mail will send to player
                 qinfo->RewMailDelaySecs = 0;                // no mail will send to player
             }
-            else if (usedMailTemplates.find(mailTemplateId) != usedMailTemplates.end())
+            else if (auto usedMtItr = usedMailTemplates.find(mailTemplateId); usedMtItr != usedMailTemplates.end())
             {
-                std::map<uint32, uint32>::const_iterator used_mt_itr = usedMailTemplates.find(mailTemplateId);
                 sLog.outErrorDb("Quest %u has `RewMailTemplateId` = %u but mail template  %u already used for quest %u, quest will not have a mail reward.",
-                                qinfo->GetQuestId(), mailTemplateId, mailTemplateId, used_mt_itr->second);
+                                qinfo->GetQuestId(), mailTemplateId, mailTemplateId, usedMtItr->second);
                 qinfo->RewMailTemplateId = 0;               // no mail will send to player
                 qinfo->RewMailDelaySecs = 0;                // no mail will send to player
             }
             else
-                usedMailTemplates[mailTemplateId] = qinfo->GetQuestId();
+                usedMailTemplates.emplace(mailTemplateId, qinfo->GetQuestId());
         }
 
         if (qinfo->NextQuestInChain)
diff --git a/src/game/ScriptMgr.cpp b/src/game/ScriptMgr.cpp
index 6e90bc0..0ade459 100644
--- a/src/game/ScriptMgr.cpp
+++ b/src/game/ScriptMgr.cpp
@@ -1243,12 +1243,7 @@ void ScriptMgr::LoadScripts(ScriptMapMap& scripts, const char* tablename)
             }
         }
 
-        if (scripts.find(tmp.id) == scripts.end())
-        {
-            ScriptMap emptyMap;
-            scripts[tmp.id] = emptyMap;
-        }
-        scripts[tmp.id].insert(ScriptMap::value_type(tmp.delay, tmp));
+        scripts[tmp.id].emplace(tmp.delay, tmp);
     }
     while (result->NextRow());
 
diff --git a/src/game/Spells/SpellMgr.cpp b/src/game/Spells/SpellMgr.cpp
index 6d648c6..5512c79 100644
--- a/src/game/Spells/SpellMgr.cpp
+++ b/src/game/Spells/SpellMgr.cpp
@@ -188,13 +188,11 @@ struct DoSpellProcEvent
     {
         SpellProcEventEntry const& spe = state->second;
         // add ranks only for not filled data (some ranks have ppm data different for ranks for example)
-        SpellProcEventMap::const_iterator spellItr = spe_map.find(spell_id);
-        if (spellItr == spe_map.end())
-            spe_map[spell_id] = spe;
+        auto emplaceResult = spe_map.emplace(spell_id, spe);
         // if custom rank data added then it must be same except ppm
-        else
+        if (!emplaceResult.second)
         {
-            SpellProcEventEntry const& r_spe = spellItr->second;
+            SpellProcEventEntry const& r_spe = emplaceResult.first->second;
             if (spe.schoolMask != r_spe.schoolMask)
                 sLog.outErrorDb("Spell %u listed in `spell_proc_event` as custom rank have different schoolMask from first rank in chain", spell_id);
 
@@ -734,14 +732,12 @@ struct DoSpellThreat
     {
         SpellThreatEntry const &ste = state->second;
         // add ranks only for not filled data (spells adding flat threat are usually different for ranks)
-        SpellThreatMap::const_iterator spellItr = threatMap.find(spell_id);
-        if (spellItr == threatMap.end())
-            threatMap[spell_id] = ste;
+        auto emplaceResult = threatMap.emplace(spell_id, ste);
 
         // just assert that entry is not redundant
-        else
+        if (!emplaceResult.second)
         {
-            SpellThreatEntry const& r_ste = spellItr->second;
+            SpellThreatEntry const& r_ste = emplaceResult.first->second;
             if (ste.threat == r_ste.threat && ste.multiplier == r_ste.multiplier && ste.ap_bonus == r_ste.ap_bonus)
                 sLog.outErrorDb("Spell %u listed in `spell_threat` as custom rank has same data as Rank 1, so redundant", spell_id);
         }
diff --git a/src/shared/Database/AutoUpdater.cpp b/src/shared/Database/AutoUpdater.cpp
index 6d5350e..345a67e 100644
--- a/src/shared/Database/AutoUpdater.cpp
+++ b/src/shared/Database/AutoUpdater.cpp
@@ -184,18 +184,18 @@ namespace DBUpdater
             const auto& migrationKey = fileMigrationPair.first;
             const auto& migration = fileMigrationPair.second;
 
-            if (dbMigrations.find(migrationKey) != dbMigrations.end()) // already applied, skip
+            auto dbMigrationItr = dbMigrations.find(migrationKey);
+            if (dbMigrationItr != dbMigrations.end()) // already applied, skip
             {
-                if (migration.Name != dbMigrations[migrationKey].Name)
+                if (migration.Name != dbMigrationItr->second.Name)
                     sLog.outInfo("[DB Auto-Updater] Migration with hash %s%s was migrated with name %s but now has name %s.", migration.Hash.c_str(),
-                        ModuleLogSuffix(migration.Module).c_str(), dbMigrations[migrationKey].Name.c_str(), migration.Name.c_str());
+                        ModuleLogSuffix(migration.Module).c_str(), dbMigrationItr->second.Name.c_str(), migration.Name.c_str());
 
-                dbMigrations.erase(migrationKey);
+                dbMigrations.erase(dbMigrationItr);
                 continue;
             }
 
             updates.push_back(std::move(fileMigrationPair.second));
-            dbMigrations.erase(migrationKey);
         }
 
 
