diff --git a/src/game/Database/DBCStructure.h b/src/game/Database/DBCStructure.h
index 7de9228..6d81e55 100644
--- a/src/game/Database/DBCStructure.h
+++ b/src/game/Database/DBCStructure.h
@@ -771,7 +771,7 @@ struct WorldSafeLocsEntry
 typedef std::set<uint32> SpellCategorySet;
 typedef std::unordered_map<uint32,SpellCategorySet > SpellCategoriesStore;
 typedef std::set<uint32> PetFamilySpellsSet;
-typedef std::map<uint32,PetFamilySpellsSet > PetFamilySpellsStore;
+typedef std::unordered_map<uint32,PetFamilySpellsSet > PetFamilySpellsStore;
 
 // Structures not used for casting to loaded DBC data and not required then packing
 struct TalentSpellPos
