diff --git a/src/game/GameEventMgr.cpp b/src/game/GameEventMgr.cpp
index 2a9af0d..cb6e0e9 100644
--- a/src/game/GameEventMgr.cpp
+++ b/src/game/GameEventMgr.cpp
@@ -252,8 +252,8 @@ void GameEventMgr::LoadFromDB()
     LoadHardcodedEvents(mGameEventHardcodedList);
 
     std::map<uint16, int16> pool2event;                     // for check unique spawn event associated with pool
-    std::map<uint32, int16> creature2event;                 // for check unique spawn event associated with creature
-    std::map<uint32, int16> go2event;                       // for check unique spawn event associated with gameobject
+    PoolEventGuidMap creature2event;                 // for check unique spawn event associated with creature
+    PoolEventGuidMap go2event;                       // for check unique spawn event associated with gameobject
 
     // list only positive event top pools, filled at creature/gameobject loading
     mGameEventSpawnPoolIds.resize(mGameEvent.size());
diff --git a/src/game/Maps/Pool/PoolManager.cpp b/src/game/Maps/Pool/PoolManager.cpp
index 54a85f3..457f4d0 100644
--- a/src/game/Maps/Pool/PoolManager.cpp
+++ b/src/game/Maps/Pool/PoolManager.cpp
@@ -129,23 +129,23 @@ void SpawnedPoolData::RemoveSpawn<Pool>(uint32 sub_pool_id, uint32 pool_id)
 ////////////////////////////////////////////////////////////
 // Methods of class PoolObject
 template<>
-void PoolObject::CheckEventLinkAndReport<Creature>(uint32 poolId, int16 event_id, std::map<uint32, int16> const& creature2event, std::map<uint32, int16> const& /*go2event*/) const
+void PoolObject::CheckEventLinkAndReport<Creature>(uint32 poolId, int16 event_id, PoolEventGuidMap const& creature2event, PoolEventGuidMap const& /*go2event*/) const
 {
-    std::map<uint32, int16>::const_iterator itr = creature2event.find(guid);
+    PoolEventGuidMap::const_iterator itr = creature2event.find(guid);
     if (itr == creature2event.end() || itr->second != event_id)
         sLog.outErrorDb("Creature (GUID: %u) expected to be listed in `game_event_creature` for event %u as part pool %u", guid, event_id, poolId);
 }
 
 template<>
-void PoolObject::CheckEventLinkAndReport<GameObject>(uint32 poolId, int16 event_id, std::map<uint32, int16> const& /*creature2event*/, std::map<uint32, int16> const& go2event) const
+void PoolObject::CheckEventLinkAndReport<GameObject>(uint32 poolId, int16 event_id, PoolEventGuidMap const& /*creature2event*/, PoolEventGuidMap const& go2event) const
 {
-    std::map<uint32, int16>::const_iterator itr = go2event.find(guid);
+    PoolEventGuidMap::const_iterator itr = go2event.find(guid);
     if (itr == go2event.end() || itr->second != event_id)
         sLog.outErrorDb("Gameobject (GUID: %u) expected to be listed in `game_event_gameobject` for event %u as part pool %u", guid, event_id, poolId);
 }
 
 template<>
-void PoolObject::CheckEventLinkAndReport<Pool>(uint32 /*poolId*/, int16 event_id, std::map<uint32, int16> const& creature2event, std::map<uint32, int16> const& go2event) const
+void PoolObject::CheckEventLinkAndReport<Pool>(uint32 /*poolId*/, int16 event_id, PoolEventGuidMap const& creature2event, PoolEventGuidMap const& go2event) const
 {
     sPoolMgr.CheckEventLinkAndReport(guid, event_id, creature2event, go2event);
 }
@@ -180,7 +180,7 @@ bool PoolGroup<T>::CheckPool() const
 
 // Method to check event linking
 template <class T>
-void PoolGroup<T>::CheckEventLinkAndReport(int16 event_id, std::map<uint32, int16> const& creature2event, std::map<uint32, int16> const& go2event) const
+void PoolGroup<T>::CheckEventLinkAndReport(int16 event_id, PoolEventGuidMap const& creature2event, PoolEventGuidMap const& go2event) const
 {
     for (uint32 i = 0; i < EqualChanced.size(); ++i)
         EqualChanced[i].template CheckEventLinkAndReport<T>(poolId, event_id, creature2event, go2event);
@@ -995,7 +995,7 @@ bool PoolManager::CheckPool(uint16 pool_id) const
 }
 
 // Method that check linking all elements to event
-void PoolManager::CheckEventLinkAndReport(uint16 pool_id, int16 event_id, std::map<uint32, int16> const& creature2event, std::map<uint32, int16> const& go2event) const
+void PoolManager::CheckEventLinkAndReport(uint16 pool_id, int16 event_id, PoolEventGuidMap const& creature2event, PoolEventGuidMap const& go2event) const
 {
     mPoolGameobjectGroups[pool_id].CheckEventLinkAndReport(event_id, creature2event, go2event);
     mPoolCreatureGroups[pool_id].CheckEventLinkAndReport(event_id, creature2event, go2event);
diff --git a/src/game/Maps/Pool/PoolManager.h b/src/game/Maps/Pool/PoolManager.h
index 2724e51..dcfa0aa 100644
--- a/src/game/Maps/Pool/PoolManager.h
+++ b/src/game/Maps/Pool/PoolManager.h
@@ -27,6 +27,7 @@
 #include "Policies/Singleton.h"
 #include "Creature.h"
 #include "GameObject.h"
+#include <unordered_map>
 
 class MapPersistentState;
 struct MapEntry;
@@ -60,6 +61,8 @@ struct PoolTemplateData
     uint32 GetSpawnCount() const;
 };
 
+typedef std::unordered_map<uint32, int16> PoolEventGuidMap;
+
 struct PoolObject
 {
     enum ObjectFlags
@@ -75,7 +78,7 @@ struct PoolObject
     bool CanBeSpawned() const;
 
     template<typename T>
-    void CheckEventLinkAndReport(uint32 poolId, int16 event_id, std::map<uint32, int16> const& creature2event, std::map<uint32, int16> const& go2event) const;
+    void CheckEventLinkAndReport(uint32 poolId, int16 event_id, PoolEventGuidMap const& creature2event, PoolEventGuidMap const& go2event) const;
 };
 
 class Pool                                                  // for Pool of Pool case
@@ -126,7 +129,7 @@ class PoolGroup
         bool isEmpty() const { return ExplicitlyChanced.empty() && EqualChanced.empty(); }
         void AddEntry(PoolObject& poolitem, uint32 maxentries);
         bool CheckPool() const;
-        void CheckEventLinkAndReport(int16 event_id, std::map<uint32, int16> const& creature2event, std::map<uint32, int16> const& go2event) const;
+        void CheckEventLinkAndReport(int16 event_id, PoolEventGuidMap const& creature2event, PoolEventGuidMap const& go2event) const;
         PoolObject* RollOne(SpawnedPoolData& spawns, uint32 triggerFrom);
         void DespawnObject(MapPersistentState& mapState, uint32 guid=0);
         void Despawn1Object(MapPersistentState& mapState, uint32 guid);
@@ -182,7 +185,7 @@ class PoolManager
         void SetExcludeObject(uint16 pool_id, uint32 db_guid_or_pool_id, bool state);
 
         bool CheckPool(uint16 pool_id) const;
-        void CheckEventLinkAndReport(uint16 pool_id, int16 event_id, std::map<uint32, int16> const& creature2event, std::map<uint32, int16> const& go2event) const;
+        void CheckEventLinkAndReport(uint16 pool_id, int16 event_id, PoolEventGuidMap const& creature2event, PoolEventGuidMap const& go2event) const;
 
         void SpawnPool(MapPersistentState& mapState, uint16 pool_id, bool instantly);
         void DespawnPool(MapPersistentState& mapState, uint16 pool_id);
