diff --git a/src/game/PacketBroadcast/PlayerBroadcaster.cpp b/src/game/PacketBroadcast/PlayerBroadcaster.cpp
index f428463..6970d2b 100644
--- a/src/game/PacketBroadcast/PlayerBroadcaster.cpp
+++ b/src/game/PacketBroadcast/PlayerBroadcaster.cpp
@@ -67,23 +67,27 @@ void PlayerBroadcaster::ProcessQueue(uint32& num_packets)
         queue.swap(m_queue);
     }
 
+    std::vector<std::pair<ObjectGuid, std::shared_ptr<PlayerBroadcaster>>> listeners;
     {
         std::lock_guard<std::mutex> guard(m_listeners_lock);
-        lastUpdatePackets = queue.size() * m_listeners.size();
-        num_packets += lastUpdatePackets;
+        listeners.reserve(m_listeners.size());
+        listeners.insert(listeners.end(), m_listeners.begin(), m_listeners.end());
+    }
 
-        for (auto& data : queue)
-        {
-            if (data.sendToSelf && data.except != GetGUID())
-                SendPacket(data.packet);
+    lastUpdatePackets = queue.size() * listeners.size();
+    num_packets += lastUpdatePackets;
 
-            for (const auto& itr : m_listeners)
-            {
-                if (itr.first == data.except)
-                    continue;
+    for (auto& data : queue)
+    {
+        if (data.sendToSelf && data.except != GetGUID())
+            SendPacket(data.packet);
+
+        for (const auto& itr : listeners)
+        {
+            if (itr.first == data.except)
+                continue;
 
-                itr.second->SendPacket(data.packet);
-            }
+            itr.second->SendPacket(data.packet);
         }
     }
 
