diff --git a/src/shared/ByteBuffer.h b/src/shared/ByteBuffer.h
index 34a5dd8..23cca47 100644
--- a/src/shared/ByteBuffer.h
+++ b/src/shared/ByteBuffer.h
@@ -566,7 +566,6 @@ inline ByteBuffer &operator>>(ByteBuffer &b, std::vector<T> &v)
     uint32 vsize;
     b >> vsize;
     v.clear();
-    v.reserve(vsize);
     while(vsize--)
     {
         T t;
@@ -593,7 +592,6 @@ inline ByteBuffer &operator>>(ByteBuffer &b, std::list<T> &v)
     uint32 vsize;
     b >> vsize;
     v.clear();
-    v.reserve(vsize);
     while(vsize--)
     {
         T t;
